### ISSUE TITLE
Feature Request: zweiter Inverter

### ISSUE THREAD

**Issue by Rayn0r** [Selbstbau-PV/Selbstbau-PV-Hoymiles-nulleinspeisung-mit-OpenDTU-und-Shelly3EM#6]
In Ticket #4 fragte @karrenbauer 

> wie bekomme ich Daten von WR2?

Hier die Antwort:

Die Daten des zweiten Inverters stehen schon in der Variable `r` drin.
Schau mal in die [WebAPI ](https://github.com/tbnobody/OpenDTU/blob/master/docs/Web-API.md#get-current-livedata) von OpenDTU. Hier siehst Du die Struktur vom JSON. Die Inverter werden dort im "Array" Inverters[] nacheinander aufgeführt.
Du kannst also die Werte des zweiten Inverters auslesen, indem Du bei `['inverters'][0]` statt der `0` eine `1` einträgst.

```
        reachable_2   = r['inverters'][1]['reachable'] # ist DTU erreichbar ?
        producing_2   = int(r['inverters'][1]['producing']) # produziert der Wechselrichter etwas ?
        altes_limit_2 = int(r['inverters'][1]['limit_absolute']) # wo war das alte Limit gesetzt
        power_dc _2   = r['inverters'][1]['AC']['0']['Power DC']['v']  # Lieferung DC vom Panel
        power_2      = r['inverters'][1]['AC']['0']['Power']['v'] # Abgabe BKW AC in **Watt**
```
Jetzt könnte zum Problem werden, wenn OpenDTU die Inverter nicht immer in der gleichen Reihenfolge liefert. Wenn Du die Inverter richtig benannt hast, sollte es allerdings kein Problem sein.
Den Namen sollte man so auslesen können:
```
        name_2 = r['inverters'][1]['name'] # Wie ist der Name?
```

**Comment by karrenbauer** [comment]
Vielen Dank! genau das habe ich gesucht!!

**Comment by Rayn0r** [comment]
Mittels:
```
    for inverter in r['inverters']:
        print(f"Name: {inverter['name']}")
        print(f"Reachable: {inverter['reachable']}\n")
```
sollte Dir das Script die Name der Inverter und deren Verfügbarkeit auch ausgeben.

**Comment by Rayn0r** [comment]
@Xenon-s 
Nein, die 300W im Script sind nicht die Ursache (also nur bedingt).

Für das Setzen des Limits muss man die Seriennummer des Inverters mitgeben, für welchen man das Limit anpassen will.
Siehe Zeile 62:
https://github.com/Selbstbau-PV/Selbstbau-PV-Hoymiles-nulleinspeisung-mit-OpenDTU-und-Shelly3EM/blob/f22b135152474c3655e9937ae26617f8cfbb1979/nulleinspeisung.py#L62
Die Erklärung gibts [hier](https://github.com/tbnobody/OpenDTU/blob/master/docs/Web-API.md) unter "Example 2: change power limit".
Wenn Du also die Leistung des 2. Inverters anpassen möchtest, dann benötigst Du im Skript eine 2. Variable für die Seriennummer des 2. Geräts. Dann muss der Request ab Zeile 59 für den 2. Inverters umgeschrieben werden. 

...und mit Sicherheit muss noch etwas mehr angepasst werden, weil ja auch hinterlegt werden muss welcher Inverter welche maximale Leistung bringen darf/kann.

**Comment by Xenon-s** [comment]
Da der zweite eh nur 400W kann, lasse ich den unangetastet und Decke damit dann alles ab. Sollte mehr benötigt werden, soll die 600W Anlage dazuschalten. Das ist ja mit dem Skript hier problemlos möglich

**Comment by karrenbauer** [comment]
ich habe mittlerweile 4 HM-Wechselrichter laufen, ich benutze für jeden ein separates Skript, das geht auch.

**Comment by Xenon-s** [comment]
Habe ich auch gemacht. Nach nicht mal einer Minute waren die Wechselrichter nicht mehr erreichbar

**Comment by StLore** [comment]
Hi, 
ich habe 2 HM bei mir im Einsatz und hatte ähnliche Effekte. 
5 Sekunden für "wait" ist zu wenig. Mit 10-20 Sekunden strapaziert man die Kommunikation nicht so sehr.
Mit 20 Sekunden habe ich keinerlei Unterbrechungen mehr.

**Comment by Xenon-s** [comment]
Danke dir für die Antwort. Sowas hatte ich mir auch gedacht. Da ich aber iobroker als Smarthome System im Einsatz habe, werde ich mir da was in Javascript bauen. Das Python Skript funktioniert sehr gut, aber dafür müsste ich einen axtra Container aufsetzen. Aber die Idee dahinter ist gut. Würde ich mich mit Python gut auskennen, würde ich es auf X belieben Inverter erweitern. Aber da findet sich bestimmt jemand :)
